torch/cuda/__init__.py

```diff
@@ -506,10 +506,13 @@ def _strtoul(s: str) -> int:
 
     # CUDA_VISIBLE_DEVICES uses something like strtoul
     # which makes `1gpu2,2ampere` is equivalent to `1,2`
-    rc: Set[int] = set()
+    rv: List[int] = []
     for elem in var.split(","):
-        rc.add(_strtoul(elem.strip()))
-    return rc
+        idx = _strtoul(elem.strip())
+        if idx in rv:
+            continue
+        rv.append(idx)
+    return rv
```

Comment:
@XuehaiPan I've confirmed that this works also with settings like `CUDA_VISIBLE_DEVICES=1,0`

Reply:
I think it's fine to apply this change in this PR. Further bug fixes may be done in #92315. cc @malfet

- #92315